k2/csrc/fsa_utils.cu

```diff
@@ -99,75 +99,237 @@ static void SplitStringToVector(const std::string &in, const char *delim,
+   format:
+
+   src_state1 dest_state1 label1 aux_label1 score1
+   src_state2 dest_state2 label2 aux_label2 score2
+   ... ...
+   final_state
+
+   The source states will be in non-descending order, and the final state does
+   not bear a cost/score -- we put the cost/score on the arc that connects to
+   the final state and set its label to -1.
+
+   @param [in]  is    The input stream that contains the transducer.
+
+   @return It returns an Fsa on CPU.
+*/
+static Fsa K2TransducerFromStream(std::istringstream &is,
+                                  Array1<int32_t> *aux_labels) {
+  K2_CHECK(aux_labels != nullptr);
+
+  std::vector<int32_t> aux_labels_internal;
+  std::vector<Arc> arcs;
+  std::vector<std::string> splits;
+  std::string line;
+
+  bool finished = false;  // when the final state is read, set it to true.
+  while (std::getline(is, line)) {
+    SplitStringToVector(line, kDelim,
+                        &splits);  // splits is cleared in the function
+    if (splits.empty()) continue;  // this is an empty line
+
+    K2_CHECK_EQ(finished, false);
+
+    auto num_fields = splits.size();
+    if (num_fields == 5u) {
+      //   0           1         2         3        4
+      // src_state  dest_state label   aux_label  score
+      int32_t src_state = StringToInt(splits[0]);
+      int32_t dest_state = StringToInt(splits[1]);
+      int32_t symbol = StringToInt(splits[2]);
+      int32_t aux_label = StringToInt(splits[3]);
+      float score = StringToFloat(splits[4]);
+      arcs.emplace_back(src_state, dest_state, symbol, score);
+      aux_labels_internal.push_back(aux_label);
+    } else if (num_fields == 1u) {
+      //   0
+      // final_state
+      (void)StringToInt(splits[0]);
+      finished = true;               // set finish
+    } else {
+      K2_LOG(FATAL) << "Invalid line: " << line
+                    << "\nK2 transducer expects a line with 1 (final_state) or "
+                    << "4 (src_state dest_state label aux_label score) fields";
+    }
+  }
+
+  K2_CHECK_EQ(finished, true) << "The last line should be the final state";
+
+  auto cpu_context = GetCpuContext();
+  *aux_labels = Array1<int32_t>(cpu_context, aux_labels_internal);
+  Array1<Arc> array(cpu_context, arcs);
+
+  bool error = true;
+  auto fsa = FsaFromArray1(array, &error);
+  K2_CHECK_EQ(error, false);
+
+  return fsa;
+}
+
+/* Create an acceptor from a stream, assuming the acceptor is in the OpenFST
+   format:
+
+   src_state1 dest_state1 label1 score1
+   src_state2 dest_state2 label2 score2
+   ... ...
+   final_state final_score
+
+   We will negate the cost/score when we read them in. Also note, OpenFST may
+   omit the cost/score if it is 0.0.
+
+   If there are multiple final states, we will create a super final state, and
+   create arcs from the old final states to the new super final state, with the
+   (negated) old final state cost/score as its cost/score, and -1 as its label.
```

Comment:
0 as its aux_label.

Reply:
We don't have aux_label for acceptors. I do have the comment for transducer, see here: https://github.com/k2-fsa/k2/blob/04872431289b5b3e9c154ef389d094d08a1a6e9c/k2/csrc/fsa_utils.cu#L356